src/pymor/analyticalproblems/functions.py

```diff
@@ -62,21 +62,18 @@ def _add_sub(self, other, sign):
                 return self
             other = ConstantFunction(other, dim_domain=self.dim_domain)
 
-        if self.name != 'LincombFunction':
-            if other.name == 'LincombFunction':
-                assert isinstance(other, LincombFunction)
+        if self.name != 'LincombFunction' or (self.name == 'LincombFunction'
+                                              and not isinstance(self, LincombFunction)):
```

Comment:
I think this is the same as
```suggestion
        if self.name != 'LincombFunction' or not isinstance(self, LincombFunction):
```
However, you're version is probably easier to understand though.